pandas/core/computation/align.py

```diff
@@ -33,9 +33,9 @@
 
 def _align_core_single_unary_op(
     term,
-) -> tuple[partial | type[FrameOrSeries], dict[str, Index] | None]:
+) -> tuple[partial | type[NDFrameT], dict[str, Index] | None]:
 
-    typ: partial | type[FrameOrSeries]
+    typ: partial | type[NDFrameT]
```

Comment:
can we avoid the variable type annotation by annotating term in the function signature? I'm not sure that the typevar is yet needed unless term is Generic.

Reply:
We need the type annotation: The variable annotation is for the return value, which in this case can have two different types. But we can replace `NDFrameT` with `NDFrame`.

Reply:
ahh yes, this is needed to prevent redefinition error. That is indeed a legimate use case of a variable type annotation to avoid refactoring the code.

In this case, I'm not sure that the type annotation being `partial | type[NDFrameT]`  is particularly useful though other than to appease mypy. This could be a good use case for Protocol, it we only need a constructor type. Not for this PR though.

again quoting https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#conventions

> avoid union return types: https://github.com/python/mypy/issues/1693